File: frontend/src/types/execucao.ts
export type BiometriaStatus = 'nao_verificado' | 'verificado' | 'falha';
export type StatusExecucao = 'pendente' | 'realizada' | 'cancelada';
export type TipoExecucao = 'presencial' | 'teleconsulta' | null;

export const STATUS_MAPPING: Record<string, StatusExecucao> = {
    "em_analise": "pendente",
    "aprovada": "realizada",
    "rejeitada": "cancelada"
};

// Interface para procedimentos na execução
export interface ProcedimentoExecucao {
    id?: string;
    procedimento_id: string;
    codigo?: string;
    nome?: string;
    quantidade: number;
    valor_unitario?: number;
    valor_total?: number;
    observacoes?: string;
}

// Interface para anexos na execução
export interface AnexoExecucao {
    id?: string;
    arquivo_id: string;
    nome_arquivo?: string;
    tipo_arquivo?: string;
    url?: string;
    tamanho?: number;
    data_upload?: string;
}

export interface Execucao {
    // Campos de identificação
    id: string;
    sessao_id: string | null;
    guia_id: string;
    numero_guia?: string;
    codigo_ficha?: string;
    codigo_ficha_temp?: boolean;
    ordem_execucao?: number | null;
    
    // Campos de paciente
    paciente_id: string | null;
    paciente_nome?: string;
    paciente_carteirinha?: string;
    
    // Campos de profissional
    profissional_id: string | null;
    profissional_executante?: string;
    usuario_executante?: string | null;
    conselho_profissional?: string | null;
    numero_conselho?: string | null;
    uf_conselho?: string | null;
    codigo_cbo?: string | null;
    
    // Campos de data e hora
    data_execucao: string;
    data_atendimento?: string | null;
    hora_inicio?: string | null;
    hora_fim?: string | null;
    
    // Campos de status e origem
    status: StatusExecucao;
    status_biometria?: BiometriaStatus;
    origem?: string;
    ip_origem?: string | null;
    tipo_execucao: TipoExecucao;
    
    // Campos de conteúdo
    procedimentos: ProcedimentoExecucao[];
    valor_total?: number | null;
    assinatura_paciente?: string | null;
    assinatura_profissional?: string | null;
    observacoes?: string | null;
    anexos: AnexoExecucao[];
    
    // Campos de auditoria
    created_at: string;
    updated_at: string;
    created_by?: string | null;
    updated_by?: string | null;
    deleted_at?: string | null;
}

export interface ExecucaoCreate extends Omit<Execucao,
    'id' | 'created_at' | 'updated_at' | 'created_by' | 'updated_by' | 'deleted_at'> {
    // Campos adicionais específicos para criação, se necessário
}

export interface ExecucaoUpdate extends Partial<ExecucaoCreate> {
    // Campos adicionais específicos para atualização, se necessário
}

export interface ExecucaoFormData {
    sessao_id: string;
    guia_id: string;
    paciente_id: string;
    data_execucao: string;
    hora_inicio?: string;
    hora_fim?: string;
    status?: StatusExecucao;
    tipo_execucao: TipoExecucao;
    procedimentos?: ProcedimentoExecucao[];
    valor_total?: number;
    assinatura_paciente?: string;
    assinatura_profissional?: string;
    observacoes?: string;
    anexos?: AnexoExecucao[];
}

// Funções auxiliares para mapeamento de valores legados
export const mapStatus = (status: string): StatusExecucao => {
    return STATUS_MAPPING[status] || status as StatusExecucao;
};

// Tipo para a resposta paginada de execuções
import { PaginatedResponse } from '@/types/api';
export type ExecucoesResponse = PaginatedResponse<Execucao>;

File: backend/schemas/execucao.py
from datetime import datetime, date
from typing import Optional
from uuid import UUID
from pydantic import BaseModel
from enum import Enum

class StatusBiometria(str, Enum):
    NAO_VERIFICADO = "nao_verificado"
    VERIFICADO = "verificado"
    FALHA = "falha"

class ExecucaoBase(BaseModel):
    """Modelo base para Execucao"""
    guia_id: UUID
    sessao_id: UUID
    data_execucao: date
    data_atendimento: Optional[date] = None
    paciente_nome: str
    paciente_carteirinha: str
    numero_guia: str
    codigo_ficha: str
    codigo_ficha_temp: bool = False
    usuario_executante: Optional[UUID] = None
    origem: str = "manual"
    ip_origem: Optional[str] = None
    ordem_execucao: Optional[int] = None
    status_biometria: StatusBiometria = StatusBiometria.NAO_VERIFICADO
    conselho_profissional: Optional[str] = None
    numero_conselho: Optional[str] = None
    uf_conselho: Optional[str] = None
    codigo_cbo: Optional[str] = None
    profissional_executante: Optional[str] = None

class ExecucaoCreate(ExecucaoBase):
    pass

class ExecucaoUpdate(BaseModel):
    guia_id: Optional[UUID] = None
    sessao_id: Optional[UUID] = None
    data_execucao: Optional[date] = None
    data_atendimento: Optional[date] = None
    paciente_nome: Optional[str] = None
    paciente_carteirinha: Optional[str] = None
    numero_guia: Optional[str] = None
    codigo_ficha: Optional[str] = None
    codigo_ficha_temp: Optional[bool] = None
    usuario_executante: Optional[UUID] = None
    origem: Optional[str] = None
    ip_origem: Optional[str] = None
    ordem_execucao: Optional[int] = None
    status_biometria: Optional[StatusBiometria] = None
    conselho_profissional: Optional[str] = None
    numero_conselho: Optional[str] = None
    uf_conselho: Optional[str] = None
    codigo_cbo: Optional[str] = None
    profissional_executante: Optional[str] = None

class Execucao(ExecucaoBase):
    """Modelo para a entidade Execucao"""
    id: UUID
    created_at: datetime
    updated_at: datetime
    deleted_at: Optional[datetime] = None
    created_by: Optional[UUID] = None
    updated_by: Optional[UUID] = None

    class Config:
        from_attributes = True